lib/image-stream/utils.js

```diff
@@ -18,45 +18,40 @@
 
 const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const fileType = require('file-type');
+const mime = require('mime-types');
+const fs = require('fs');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
+exports.getArchiveMimeType = (filename) => {
+  if (mime.lookup(filename)) {
+    return Bluebird.resolve(mime.lookup(filename));
+  }
 
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
+  const FILE_TYPE_ID_BYTES = 261;
 
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
+  return Bluebird.using(fs.openAsync(filename, 'r').disposer((fd) => {
+    return fs.closeAsync(fd);
+  }), (fd) => {
     const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+    const buffer = Buffer.alloc(FILE_TYPE_ID_BYTES);
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
+    return fs.readAsync(fd, buffer, BUFFER_START, FILE_TYPE_ID_BYTES, null).then(() => {
+      return _.get(fileType(buffer), [ 'mime' ], 'application/octet-stream');
```

Comment:
Lets either:

- Bring back the old behavior (only return supported MIME types or `application/octet-stream`), or
- Allow a `The ${type} format is not supported` error, but make it a user error, so that it doesn't get to TrackJS/Sentry